src/shadowbox/model/metrics.ts

```diff
@@ -28,3 +28,15 @@ export interface DataUsageByUser {
 export interface DataUsageTimeframe {
   hours: number;
 }
+
+
+export interface TunneltimeMetrics {
+  access_key?: Map<string, number>;
+  country?: Map<string, number>;
+  asn?: Map<string, number>;
+}
+
+export interface TunneltimeQuery {
```

Comment:
The request shape looks mostly good. I would use a duration type, and not call it hours. Perhaps `time_window` or `since_time_ago` kind of concept.

For the group name. Replace it with a list of group by fields. Alternatively we can return all dimensions always, but i like the idea of specifying the group by dimensions, as it can be a big optimization for large servers.